src/ring_theory/roots_of_unity.lean

```diff
@@ -715,4 +716,79 @@ lemma nth_roots_one_eq_bind_primitive_roots {ζ : R} {n : ℕ} (hpos : 0 < n)
 
 end integral_domain
 
+section minimal_polynomial
+
+open minimal_polynomial
+
+variables {n : ℕ} {K : Type*} [field K] {μ : K} (h : is_primitive_root μ n) (hpos : 0 < n)
+
+include n μ h hpos
+
+/--`μ` is integral over `ℤ`. -/
+lemma is_integral : is_integral ℤ μ :=
+begin
+  use (X ^ n - 1),
+  split,
+  { exact (monic_X_pow_sub_C 1 (ne_of_lt hpos).symm) },
+  simp only [((is_primitive_root.iff_def μ n).mp h).left, eval₂_one, eval₂_X_pow, eval₂_sub,
+  sub_self],
+end
+
+/--The minimal polynomial of `μ` is monic. -/
+lemma is_monic_minimal_polynomial : (minimal_polynomial (is_integral h hpos)).monic :=
+(minimal_polynomial.monic (is_integral h hpos))
+
+/--The minimal polynomial of `μ` is primitive. -/
+lemma is_primitive_minimal_polynomial : is_primitive
+(minimal_polynomial (is_integral h hpos)) := monic.is_primitive
+(is_monic_minimal_polynomial h hpos)
+
+variables [char_zero K]
+
+/--`μ ` is integral over `ℚ`. -/
+lemma is_algebraic : _root_.is_integral ℚ μ :=
+@is_integral_of_is_scalar_tower ℤ ℚ K _ _ _ (algebra_int ℚ) _ _ add_comm_group.int_is_scalar_tower
+  μ (is_integral h hpos)
+
+/--The minimal polynomial of `μ` over `ℚ` is the same as the minimal polynomial over `ℤ`. -/
+lemma min_poly_rational_eq_min_poly_integer : minimal_polynomial (is_algebraic h hpos)
```

Comment:
This lemma is not specific about primitive roots, right? Does it make sense to have a general lemma about towers `R K L`, where `R` is an integral domain, `K` its field of fractions and `L` some `K`-algebra (maybe a field?).
Then if `x : L` is integral over `R`, the result from this lemma should hold, right?